### PR TITLE
fix calculate json-anydata length error， when leaf/leaflist value con… … f0067eb …tains ‘{' or '}'

### PR DESCRIPTION
**Pull request by olderma**
rpc-message ： {"ietf-netconf-nmda:edit-data":{"datastore":"ietf-datastores:running","config":{password:"Root{{{{123"}}}

when calculate 'password' len， the result is wrong.

### CODE REVIEW

File: src/parser_json.c [hunk 1]
```diff
@@ -398,10 +398,22 @@ json_get_anydata(struct lyd_node_anydata *any, const char *data)
     do {
         switch (data[len]) {
         case '{':
-            c++;
+            if (skip == 0) {
```
Comment (michalvasko) [inline src/parser_json.c:401]: Use `!skip` condition instead.

File: src/parser_json.c [hunk 2]
```diff
@@ -398,10 +398,22 @@ json_get_anydata(struct lyd_node_anydata *any, const char *data)
...
+                c++;
+            }
             break;
         case '}':
-            c--;
+            if (skip == 0) {
```
Comment (michalvasko) [inline src/parser_json.c:406]: Replace with `!skip`.

File: src/parser_json.c [hunk 3]
```diff
@@ -398,10 +398,22 @@ json_get_anydata(struct lyd_node_anydata *any, const char *data)
...
+                c--;
+            }
+            break;
+        case '\\':
+            if (skip == 1 && data[len + 1]) {
```
Comment (michalvasko) [inline src/parser_json.c:411]: Replace `skip == 1` with just `skip` and please add a comment with an explanation. Something like `when parsing a string, ignore escaped quotes to not end it prematurely`.

File: src/parser_json.c [hunk 4]
```diff
@@ -398,10 +398,22 @@ json_get_anydata(struct lyd_node_anydata *any, const char *data)
...
+                len++;
+            }
+            break;
+        case '\"':
+            skip = (skip == 1) ? 0 : 1;
```
Comment (michalvasko) [inline src/parser_json.c:416]: Could be replaced with `skip = !skip;`.

**Review by michalvasko — CHANGES_REQUESTED** [review]

**Comment by michalvasko** [comment]
Hi,
the idea is right, this was not handled, but please keep the code clean by treating boolean values the way they should be.

**Comment by olderma** [comment]
> Hi,
> the idea is right, this was not handled, but please keep the code clean by treating boolean values the way they should be.

ok，I have fix the issues.

**Review by michalvasko — APPROVED** [review]

**Comment by michalvasko** [comment]
Okay, looks good now, thanks.

**Pull request merged**